insights/parsers/proc_keys.py

```diff
@@ -0,0 +1,91 @@
+        Usage (string):            The count of the number of kernel credential structures
+                                that are pinning the key.
+        Timeout (string):       The amount of time until the key will expire (weeks, days,
+                                hours, minutes, and seconds). The string perm here means
+                                that the key is permanent (no timeout). The string expd
+                                means that the key has already expired, but has not yet
+                                been garbage collected.
+        Permissions (string):   The key permissions, expressed as four hexadecimal bytes
+                                containing, from left to right, the possessor, user, group,
+                                and other permissions.
+        UID (string):           The user ID of the key owner.
+        GID (string):           The group ID of the key.
+        Type (string) :         The key type.
+        Description (string)    The key description (name). For most key types, it has the
+                                form name[: extra-info]. (The name subfield is the key's
+                                description (name). The optional extra-info field provides
+                                some further information about the key.)
+
+    Sample content of '/proc/keys' file looks like::
+
+        009a2028 I--Q---   1 perm 3f010000  1000  1000 user     krb_ccache:primary: 12
+        1806c4ba I--Q---   1 perm 3f010000  1000  1000 keyring  _pid: 2
+        25d3a08f I--Q---   1 perm 1f3f0000  1000 65534 keyring  _uid_ses.1000: 1
+        28576bd8 I--Q---   3 perm 3f010000  1000  1000 keyring  _krb: 1
+        2c546d21 I--Q--- 190 perm 3f030000  1000  1000 keyring  _ses: 2
+        30a4e0be I------   4   2d 1f030000  1000 65534 keyring  _persistent.1000: 1
+        32100fab I--Q---   4 perm 1f3f0000  1000 65534 keyring  _uid.1000: 2
+        32a387ea I--Q---   1 perm 3f010000  1000  1000 keyring  _pid: 2
+        3ce56aea I--Q---   5 perm 3f030000  1000  1000 keyring  _ses: 1
+
+
+    Examples:
+        >>> type(proc_keys)
+        <class 'insights.parsers.proc_keys.ProcKeys'>
+        >>> proc_keys.data[0]['id']
+        '009a2028'
+        >>> proc_keys.data[0]['flags']
+        'I--Q---'
+        >>> proc_keys.data[0]['usage']
+        '1'
+        >>> proc_keys.data[0]['timeout']
+        'perm'
+        >>> proc_keys.data[0]['permissions']
+        '3f010000'
+        >>> proc_keys.data[0]['uid']
+        '1000'
+        >>> proc_keys.data[0]['gid']
+        '1000'
+        >>> proc_keys.data[0]['type']
+        'user'
+        >>> proc_keys.data[0]['description']
+        'krb_ccache:primary: 12'
+    """
+
+    def parse_content(self, content):
+        content = get_active_lines(content, comment_char="COMMAND>")
+
+        if not content:
+            raise SkipException("No Contents")
+
+        self.data = []
```

Comment:
`self.data` is not recommended as it's too general to distinguish the type of the real `data`.

If mixed the parser with `list`, this `self.data` attribute can be saved.